sklearn/calibration.py

```diff
@@ -311,145 +370,202 @@ def _more_tags(self):
+        if n_classes == 2:
+            preds = preds[:, 1:]
+    else:
+        raise RuntimeError("'method' needs to be one of 'decision_function' "
+                           "or 'predict_proba'.")
+    return preds
 
-        idx_pos_class = self.label_encoder_.\
-            transform(self.base_estimator.classes_)
 
-        return df, idx_pos_class
+def _get_predictions(pred_method, X, label_encoder_):
+    """Returns predictions for `X` and the index of classes present.
 
-    def fit(self, X, y, sample_weight=None):
-        """Calibrate the fitted model
+    Parameters
+    ----------
+    pred_method : callable
+        Prediction method.
 
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_features)
-            Training data.
+    X : array-like
+        Data used to obtain predictions.
 
-        y : array-like, shape (n_samples,)
-            Target values.
+    label_encoder_ : LabelEncoder instance
+        LabelEncoder instance fitted on all the targets.
 
-        sample_weight : array-like of shape (n_samples,), default=None
-            Sample weights. If None, then samples are equally weighted.
+    Returns
+    -------
+    preds : array-like, shape (X.shape[0], len(clf_fitted.classes_))
+        The predictions. Note if there are 2 classes, array is of shape
+        (X.shape[0], 1).
+    """
+    preds = pred_method(X)
+    n_classes = len(label_encoder_.classes_)
+    preds = _reshape_preds(preds, pred_method.__name__, n_classes)
+    return preds
 
-        Returns
-        -------
-        self : object
-            Returns an instance of self.
-        """
 
-        self.label_encoder_ = LabelEncoder()
-        if self.classes is None:
-            self.label_encoder_.fit(y)
+def _fit_calibrator(clf_fitted, preds, y, label_encoder_, method,
+                    sample_weight=None):
+    """Fit calibrator(s) and return a `_CalibratedClassiferPipeline`
+    instance.
+
+    `n_classes` (i.e. `len(clf_fitted.classes_)`) calibrators are fitted.
+    However, if `n_classes` equals 2, one calibrator is fit.
+
+    Parameters
+    ----------
+    clf_fitted : Estimator instance
+        Fitted classifier.
+
+    preds :  array-like, shape (n_samples, n_classes)
+        Predictions for calibrating the predictions.
+        If binary, shape (n_samples, 1).
+
+    y : ndarray, shape (n_samples,)
+        The targets.
+
+    label_encoder_ : LabelEncoder instance
+        LabelEncoder instance fitted on all the targets.
+
+    method : {'sigmoid', 'isotonic'}
+        The method to use for calibration.
+
+    sample_weight : ndarray, shape (n_samples,), default=None
+        Sample weights. If None, then samples are equally weighted.
+
+    Returns
+    -------
+    pipeline : _CalibratedClassiferPipeline instance
+    """
+    Y = label_binarize(y, classes=label_encoder_.classes_)
+    pos_class_indices = label_encoder_.transform(clf_fitted.classes_)
+    calibrated_classifiers = []
```

Comment:
I think this should just be called `calibrators`.

The calibrated classifiers are what this function returns.

Reply:
good point